swagger.yaml

```diff
@@ -1,75 +1,298 @@
+      tags:
+      - "cards"
+      summary: "Retrieve cards based on search parameters"
+      description: ""
+      operationId: "getCards"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       parameters:
-        - $ref: '#/parameters/pretty'
+        - name: "id"
+          in: "query"
+          description: "id of card in database"
+          required: false
+          type: "integer"
+          format: "int64"
+
+        - name: "types"
+          in: "query"
+          description: "List of types of cards"
+          required: false
+          type: "array"
+          items:
+            type: "string"
+            enum:
+            - "attack"
+            - "skill"
+            - "power"
+            - "status"
+            - "curse"
+            default: "attack,skill,power,status,curse"
+          collectionFormat: "multi"
+
+        - name: "name"
+          in: "query"
+          description: "Card name"
+          required: false
+          type: "string"
+
+        - name: "color"
+          in: "query"
+          description: "List of colors of cards"
+          required: false
+          type: "array"
+          items:
+            type: "string"
+            enum:
+            - "red"
+            - "green"
+            - "blue"
+            - "colorless"
+            default: "red,green,blue,colorless"
+          collectionFormat: "multi"
+
+        - name: "rarity"
+          in: "query"
+          description: "List of rarities of cards"
+          required: false
+          type: "array"
+          items:
+            type: "string"
+            enum:
+            - "basic"
+            - "common"
+            - "uncommon"
+            - "rare"
+            default: "basic,common,uncommon,rare"
+          collectionFormat: "multi"
+
+        - name: "energyMin"
+          in: "query"
+          description: "Minimum value of energy costs of cards (X = 0). Used with energyMax"
+          required: false
+          type: "integer"
+
+        - name: "energyMax"
+          in: "query"
+          description: "Maximum value of energy costs of cards (X = 0). Used with energyMin"
+          required: false
+          type: "integer"
+
+        - name: "keywords"
+          in: "query"
+          description: "List of keywords found in a card's description"
+          required: false
+          type: "array"
+          items:
+            type: "string"
+
+        - name: "number"
+          in: "query"
+          description: "Number of cards to return"
+          required: false
+          type: integer
+          default: 10
+
       responses:
         200:
-          description: Object containing information
+          description: "Cards successfully retrieved"
+          schema:
+            $ref: "#/definitions/MultiData"
+        400:
+          description: "Invalid request"
+          schema:
+            $ref: "#/definitions/Error"
+        404:
```

Comment:
I'm not sure if this endpoint should return a 404. In jsonAPI spec, endpoints that usually return an array of objects would return a 200 with an empty `"data"` array if no cards were found that matched the query. http://jsonapi.org/format/#fetching-resources-responses-200